lib/nerves_hub_link/extensions/health/default_report.ex

```diff
@@ -59,6 +61,21 @@ defmodule NervesHubLink.Extensions.Health.DefaultReport do
     |> Enum.reduce(%{}, &Map.merge/2)
   end
 
+  defp get_metric_sets() do
+    Application.get_env(:nerves_hub_link, :health, [])
+    |> Keyword.get(:metric_sets, @default_metric_sets)
+    |> Enum.map(fn metric_set ->
+      if metric_set in [Default, Defaults] do
+        @default_metric_sets
+      else
+        metric_set
+      end
+    end)
+    |> List.flatten()
+    |> Enum.map(& &1.metrics())
+    |> Enum.reduce(%{}, &Map.merge/2)
```

Comment:
This value from application config + `Enum.reduce` + merging is what is being missed by dialyzer. Here we should probably be more strict in the keys merged. Mixture of atoms and strings will be duplicated over the wire, but deduped on decode on the server leading to wasted bytes

<img width="323" alt="image" src="https://github.com/user-attachments/assets/559c1e58-453b-4abe-a8af-6ecee399e5af" />

One approach would be to filter and transform as you are iterating through the metric sets instead
```suggestion
    for metric_set <- metric_sets, {k, v} <- metric_set.metrics(), into: %{}, do: {to_string(k), v}
```

<img width="771" alt="image" src="https://github.com/user-attachments/assets/6fb19bee-47d8-48c1-adc5-922eddc6485b" />

Reply:
Great find. I don't love the `for` comprehension, but I've gone with it.